src/terms/named_node.rs

```diff
@@ -0,0 +1,80 @@
+use crate::error::Error;
+use regex::Regex;
+use std::fmt::{Display, Formatter};
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct NamedNode {
+    // TODO: create a type FQDN that can convert to/from String, url, and &str
+    pub(crate) value: String, // TODO: is there a way to hide the field since I'm validating it in the `new` fn
+                              //   A: yes, in fact, you've already done this by not marking the field `pub`
+}
+
+// TODO: need a more robust regex
+const ABSOLUTE_URI_EXPRESSION: &str = r"^(?:http|https)://(?:\w+\.)+\w+/.*$";
+
+impl NamedNode {
+    pub fn new(value: &str) -> Result<Self, Error> {
+        let regex =
+            Regex::new(ABSOLUTE_URI_EXPRESSION).expect("Absolute URI Regex failed to compile");
+
+        if regex.is_match(value) {
+            Ok(NamedNode {
+                value: value.to_string(),
+            })
+        } else {
+            Err(Error::InvalidURIError(value.to_string()))
+        }
+    }
+}
+
+impl Display for NamedNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "<{}>", self.value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_equality() {
+        let nn1 = NamedNode {
+            value: "foo".to_string(),
+        };
+        let nn2 = NamedNode {
+            value: "bar".to_string(),
+        };
+        let nn3 = NamedNode {
+            value: "foo".to_string(),
+        };
+
+        assert_eq!(nn1, nn1);
+        assert_ne!(nn1, nn2);
+        assert_eq!(nn1, nn3);
+        assert_eq!(nn3, nn1);
+    }
+
+    #[test]
+    fn new_valid_url() {
+        let nn = NamedNode {
+            value: "https://foo.com/bar".to_string(),
+        };
+
+        assert_eq!(NamedNode::new("https://foo.com/bar").unwrap(), nn);
+    }
+
+    #[test]
+    fn new_invalid_url() {
+        let r = NamedNode::new("foo");
+        assert_eq!(r.is_err(), true);
```

Comment:
Using `assert_eq!` to compare something to true is non-idiomatic. You should use just the `assert!` macro in this situation:
```rust
assert!(r.is_err());
```

Reply:
good call - fixed